File: python-puka/emit_log.py
#!/usr/bin/env python
import puka
import sys

client = puka.Client("amqp://localhost/")
promise = client.connect()
client.wait(promise)


promise = client.exchange_declare(exchange='logs', type='fanout')
client.wait(promise)

message = ' '.join(sys.argv[1:]) or "info: Hello World!"
promise = client.basic_publish(exchange='logs', routing_key='', body=message)
client.wait(promise)

print " [x] Sent %r" % (message,)
client.close()

File: javascript-nodejs/src/emit_log.js
#!/usr/bin/env node

const amqp = require('amqplib');

async function main() {
    const connection = await amqp.connect(process.env.AMQP_URL || 'amqp://localhost');
    const channel = await connection.createChannel();

    const exchange = 'logs';
    const msg = process.argv.slice(2).join(' ') || 'Hello World!';

    await channel.assertExchange(exchange, 'fanout', {
        durable: false
    });
    channel.publish(exchange, '', Buffer.from(msg));
    console.log(" [x] Sent %s", msg);

    setTimeout(function() {
        connection.close();
        process.exit(0);
    }, 500);
}

main();
